File: SkyEditor.Core/IO/iSavable.vb
﻿Namespace IO
    ''' <summary>
    ''' Marks a class that supports saving.
    ''' </summary>
    Public Interface ISavable
        ''' <summary>
        ''' Saves the class to the last filename.
        ''' </summary>
        Function Save(provider As IOProvider) As Task
        ''' <summary>
        ''' Raised when the file is saved.
        ''' </summary>
        ''' <param name="sender"></param>
        ''' <param name="e"></param>
        Event FileSaved(sender As Object, e As EventArgs)
    End Interface

End Namespace

File: SkyEditor.Core/IO/ISavable.cs
using SkyEditor.IO.FileSystem;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SkyEditor.Core.IO
{
    /// <summary>
    /// Represents a class that supports saving
    /// </summary>
    public interface ISavable
    {
        /// <summary>
        /// Raised when the file or object is saved
        /// </summary>
        event EventHandler FileSaved;

        /// <summary>
        /// Saves the class
        /// </summary>
        /// <param name="provider"></param>
        /// <returns></returns>
        Task Save(IFileSystem provider);  
    }
}
